lms/djangoapps/django_comment_client/forum/views.py

```diff
@@ -258,15 +267,21 @@ def single_thread(request, course_key, discussion_id, thread_id):
     """
     Renders a response to display a single discussion thread.
     """
-
     nr_transaction = newrelic.agent.current_transaction()
 
     course = get_course_with_access(request.user, 'load_forum', course_key)
-    course_settings = make_course_settings(course)
+    course_settings = make_course_settings(course, request.user)
     cc_user = cc.User.from_django_user(request.user)
     user_info = cc_user.to_dict()
     is_moderator = cached_has_permission(request.user, "see_all_cohorts", course_key)
 
+    # Verify that the student has access to this thread if belongs to a discussion module
+    accessible_discussion_ids = [
+        module.discussion_id for module in utils.get_accessible_discussion_modules(course, request.user)
```

Comment:
P.s. I believe that these discussion ids can be extracted from the `course_settings` variable, bringing mongo calls back down. I'm looking into this right now.